### DISCUSSION TITLE
Installing a package makes homebrew update tons of dependencies and breaks my system

### DISCUSSION THREAD

**Discussion by patrick99e99** [Homebrew/discussions#1112] [Everyday usage]
I've had this happen twice so far, and am very confused and wondering how I can prevent it from happening in the future..

I has elasticsearch 6.8 installed on my machine, and today I upgraded to 7.  So I did brew uninstall elasticsearch@6..  It warned me about files still existing somewhere in /usr/local/opt/elasticsearch ... so I rm'd those...  I then did brew install elasticsearch@7 ...  It installed everything.

And then I tried to run this plugin I had, kibana..  It no longer was there...  So, I brew installed kibana...

And then it installed kibana...

And then it started going crazy, installing billions of things...  openssl...

and then eventually got to ffmpeg

... FFMPEG???  That has nothing to do with kibana!!!!

And then it bailed telling me my version of xcode is too old and to update it.  Haha..  My version of xcode is the last compatible version for this version of Mac OS 10.14.3.........

Anyway.. the next thing I see, among the many things, it apparently upgraded postgres on me, and my database is now gone.

So...  I guess my question is, why oh why does homebrew not prompt the user with:

"DO YOU WANT ME TO UPDATE EVERYTHING IN YOUR SYSTEM WHICH MIGHT COMPLETELY BREAK YOUR SYSTEM (Y/N)?"

**Comment by carlocab** [comment]
See https://github.com/Homebrew/discussions/discussions/1107, https://docs.brew.sh/FAQ#why-does-brew-upgrade-formula-also-upgrade-a-bunch-of-other-stuff.

`kibana` depends directly on `python@3.9`, and indirectly on `openssl` (through `python@3.9`). So, if you want to install `kibana`, `python@3.9`, and all formulae in both directions of `python@3.9`'s dependency graph (i.e. anything that `python@3.9` depends on, and anything that depends on `python@3.9`, recursively) needs to be up-to-date. This includes `ffmpeg`, as `ffmpeg` depends on formulae that depend on `python@3.9` and `openssl`.

**Reply by patrick99e99** [reply]
I see..  Thank you for the reply. I just wish homebrew would simply output a list of everything it's going to install and ask you: Y/N ... So I could avoid something catastrophic like what happened today.

Does brew happen to have a --dry-run flag that would do this?

**Reply by carlocab** [reply]
Indeed there is:
```
❯ brew help upgrade
Usage: brew upgrade [options] [outdated_formula|outdated_cask ...]

Upgrade outdated casks and outdated, unpinned formulae using the same options
they were originally installed with, plus any appended brew formula options. If
cask or formula are specified, upgrade only the given cask or formula
kegs (unless they are pinned; see pin, unpin).

Unless HOMEBREW_NO_INSTALL_CLEANUP is set, brew cleanup will then be run for
the upgraded formulae or, every 30 days, for all formulae.
[snip]
  -n, --dry-run                    Show what would be upgraded, but do not
                                   actually upgrade anything.
```

**Reply by patrick99e99** [reply]
But that flag isn't for install?

Today's problem came from brew install kibana, not brew upgrade...

**Reply by carlocab** [reply]
Oh, whoops, sorry. No, there isn't a dry-run flag for `install`.

**Reply by patrick99e99** [reply]
boo.... :(

**Reply by carlocab** [reply]
You could always do `brew upgrade --dry-run` to see what the worst that could happen is.

**Reply by patrick99e99** [reply]
I will get `Error: kibana not installed`

**Reply by patrick99e99** [reply]
oh sorry I misunderstood...  I see what you mean, but I don't think that tells me anything about the dependencies of individual packages.

**Reply by carlocab** [reply]
Can you do `brew uninstall kibana`?

**Reply by carlocab** [reply]
`brew deps $formula` will tell you the dependencies of a formula. You can also use `brew deps --tree $formula`.
